### PR TITLE
[Keyboard] Add city42

### PR DESCRIPTION
**Pull request by controller-works**
Adding support for city42 keyboard from Controller Works.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/controllerworks/city42/config.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+#pragma once
+
+#define WS2812_PIO_USE_PIO1
+#define RGBLED_NUM 42
+#define RGB_MATRIX_LED_COUNT RGBLED_NUM
+
+/* Top left key */
+#define BOOTMAGIC_LITE_ROW    0
+#define BOOTMAGIC_LITE_COLUMN 0
+/* Top right key on right half */
```
Comment (zvecr) [inline keyboards/controllerworks/city42/config.h:27]:
```suggestion
```

File: keyboards/controllerworks/city42/config.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+/* Cirque trackpad over SPI. Production */
```
Comment (zvecr) [inline keyboards/controllerworks/city42/config.h:29]:
```suggestion
```

File: keyboards/controllerworks/city42/keymaps/default/config.h [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#ifdef RGB_MATRIX_ENABLE
```
Comment (zvecr) [inline keyboards/controllerworks/city42/keymaps/default/config.h:19]: The config being at the keymap level should really be avoided, as it causes duplication and configurator to not work correctly.

Comment (zvecr) [reply keyboards/controllerworks/city42/keymaps/default/config.h:19]: You need to stop marking this as resolved when you have done absolutely nothing about it.

Comment (controller-works) [reply keyboards/controllerworks/city42/keymaps/default/config.h]: I have moved the RGB_MATRIX settings out of the keymaps as requested.

File: keyboards/controllerworks/city42/config.h
Comment (dunk2k) [inline keyboards/controllerworks/city42/config.h:1]: can settings in this file that have data-driven equivalents be moved to `info.json`?

File: keyboards/controllerworks/city42/readme.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# city42
+![city42](https://i.imgur.com/YZNW4EZ.jpg)
```
Comment (dunk2k) [inline keyboards/controllerworks/city42/readme.md:2]:
```suggestion
![city42](https://i.imgur.com/YZNW4EZh.jpg)
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/controllerworks/city42/info.json [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    "bootloader": "rp2040",
+    "diode_direction": "COL2ROW",
+    "build": {
+        "lto":true
+    },
+    "matrix_pins": {
+        "cols": 
+            ["GP28", "GP27", "GP26","GP25", "GP24", "GP23", "GP0", "GP1", "GP2", "GP3", "GP4", "GP5"],
+        "rows":
+            ["GP12", "GP13", "GP14", "GP15"]
```
Comment (dunk2k) [inline keyboards/controllerworks/city42/info.json:22]: keeps in uniform with everybody else 😉
```suggestion
        "cols": ["GP28", "GP27", "GP26","GP25", "GP24", "GP23", "GP0", "GP1", "GP2", "GP3", "GP4", "GP5"],
        "rows": ["GP12", "GP13", "GP14", "GP15"]
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/controllerworks/city42/config.h
Comment (drashna) [reply keyboards/controllerworks/city42/config.h:1]: `qmk migrate` can help with a majority of that

File: keyboards/controllerworks/city42/info.json [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+                { "matrix": [2, 10], "x": 12, "y": 2.4 },
+                { "matrix": [2, 11], "x": 13, "y": 2.4 },
+                { "matrix": [3, 3], "x": 3.5, "y": 3.25 },
+                { "matrix": [3, 4], "x": 4.5, "y": 3.5 },
+                { "matrix": [3, 5], "x": 5.5, "y": 3.75, "h":1.5 },
+                { "matrix": [3, 6], "x": 7.5, "y": 3.75, "h":1.5 },
+                { "matrix": [3, 7], "x": 8.5, "y": 3.5 },
+                { "matrix": [3, 8], "x": 9.5, "y": 3.25 }
+            ]
+          }
```
Comment (dunk2k) [inline keyboards/controllerworks/city42/info.json:129]:
```suggestion
        }
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/controllerworks/city42/config.h
Comment (controller-works) [reply keyboards/controllerworks/city42/config.h:1]: I removed most of the RGB Matrix configuration from the config.h. However, I cannot see how to get rid of RGB_MATRIX_LED_COUNT. Does that still have to be in the config.h?

File: keyboards/controllerworks/city42/info.json (hunk 6 above)
Comment (controller-works) [reply keyboards/controllerworks/city42/info.json]: I think this was resolved in the info.json rewrite.

File: keyboards/controllerworks/city42/config.h
Comment (dunk2k) [reply keyboards/controllerworks/city42/config.h:1]: > I removed most of the RGB Matrix configuration from the config.h. However, I cannot see how to get rid of RGB_MATRIX_LED_COUNT. Does that still have to be in the config.h?

At this current time, there isn't a data-driven equivalent for mentioned setting.

**Review by drashna — APPROVED** [review]

**Comment by controller-works** [comment]
I appreciate everyone's time in reviewing this PR. I believe all requested changes have been implemented.

**Review by dunk2k — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**